src/diffusers/loaders.py

```diff
@@ -1087,9 +1089,12 @@ def load_lora_into_text_encoder(cls, state_dict, network_alpha, text_encoder, lo
 
                 load_state_dict_results = text_encoder.load_state_dict(text_encoder_lora_state_dict, strict=False)
                 if len(load_state_dict_results.unexpected_keys) != 0:
-                    raise ValueError(
-                        f"failed to load text encoder state dict, unexpected keys: {load_state_dict_results.unexpected_keys}"
-                    )
+                    unexpected_keys = load_state_dict_results.unexpected_keys
+                    is_sdxl_ckpt = any("text_encoder_" in k for k in unexpected_keys)
+                    if not is_sdxl_ckpt:
+                        raise ValueError(
+                            f"failed to load text encoder state dict, unexpected keys: {load_state_dict_results.unexpected_keys}"
+                        )
```

Comment:
We can avoid this adhoc check for which model we're loading by passing in a dict with only the keys of the expected text_encoder :)

i.e.

```
class SDXLPipelin:
    def load_lora_weights():
        state_dict, network_alpha = self.lora_state_dict(pretrained_model_name_or_path_or_dict, **kwargs)
        self.load_lora_into_unet(state_dict, network_alpha=network_alpha, unet=self.unet)
        
        text_encoder_state_dict = { k: v for k, v in state_dict if 'text_encoder.' in k }
        self.load_lora_into_text_encoder(
            text_encoder_state_dict,
            network_alpha=network_alpha,
            text_encoder=self.text_encoder,
            prefix="text_encoder",
            lora_scale=self.lora_scale,
        )

       text_encoder_2_state_dict = { k: v for k, v in state_dict if 'text_encoder_2.' in k }
        self.load_lora_into_text_encoder(
            text_encoder_2_state_dict,
            network_alpha=network_alpha,
            text_encoder=self.text_encoder_2,
            prefix="text_encoder_2",
            lora_scale=self.lora_scale,
        )
```

We really want to as much as possible avoid checking what's in the state dict. If we can do that checking in the code specific to that particular model, we should :)